state/protocol/validity_test.go

```diff
@@ -201,6 +201,53 @@ func TestBootstrapInvalidEpochCommit(t *testing.T) {
 	})
 }
 
+// TestIsValidEpochCommitBackwardCompatible tests that the implementation is backward compatible with the previous version of EpochCommit.
+// The main difference in validation logic is that the old version requires that number of consensus participants is
+// equal to the number of keys(random beacon participants) for new version of the [flow.EpochCommit] this is not required rather
+// we rely on a threshold for random beacon participants.
+// TODO(EFM, #6794): Remove this once we complete the network upgrade
+func TestIsValidEpochCommitBackwardCompatible(t *testing.T) {
+	_, result, _ := unittest.BootstrapFixture(participants)
+	setup := result.ServiceEvents[0].Event.(*flow.EpochSetup)
+	commit := result.ServiceEvents[1].Event.(*flow.EpochCommit)
+	requiredThreshold := protocol.RandomBeaconSafetyThreshold(uint(len(commit.DKGIndexMap)))
+	require.Less(t, int(requiredThreshold), len(commit.DKGParticipantKeys),
+		"threshold has to be at lower than the number of keys, otherwise the test is invalid")
+	// sample the required threshold, since it's lower than the number of keys it will be valid for the
+	// new version of the [flow.EpochCommit] but not for the v0 version since it doesn't have the [flow.DKGIndexMap] field.
+	sampled, err := setup.Participants.Filter(filter.IsConsensusCommitteeMember).Sample(requiredThreshold)
+	require.NoError(t, err)
```

Comment:
I would move this right in front of [line 238](https://github.com/onflow/flow-go/blob/0b12b6342a5e526a6860edb07d789748e3c03f4a/state/protocol/validity_test.go#L238). This way, I feel the explanantion of the test is one step at a time